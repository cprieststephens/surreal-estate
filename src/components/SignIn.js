import React from "react";
import "../styles/sign-in.css";

const SignIn = ({ setuserID }) => {
  const handleLogin = () => {};

  return (
    <div className="sign-in">
      <form>
        <label htmlFor="sign-in__email">
          Email: <input type="email" id="sign-in__email" />
        </label>
        <label htmlFor="sign-in__password">
          Password: <input type="password" id="sign-in__password" />
        </label>
      </form>
      <button type="button" onClick={handleLogin}>
        Sign In
      </button>
    </div>
  );
};

export default SignIn;
