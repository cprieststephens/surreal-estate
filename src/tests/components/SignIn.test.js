import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import SignIn from "../../components/SignIn";

describe("SignIn", () => {
  const validProps = {
    setUserID: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SignIn setUserID={validProps.setUserID} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders an email input", () => {
    render(
      <MemoryRouter>
        <SignIn setUserID={validProps.setUserID} />
      </MemoryRouter>
    );
    const emailInput = screen.getByPlaceholderText("Enter email");

    expect(emailInput).toBeInTheDocument();
  });

  it("renders a password input", () => {
    render(
      <MemoryRouter>
        <SignIn setUserID={validProps.setUserID} />
      </MemoryRouter>
    );
    const passwordInput = screen.getByPlaceholderText("Enter password");

    expect(passwordInput).toBeInTheDocument();
  });

  it("renders a submit button", () => {
    render(
      <MemoryRouter>
        <SignIn setUserID={validProps.setUserID} />
      </MemoryRouter>
    );
    const submitButton = screen.getByRole("button");

    expect(submitButton).toBeInTheDocument();
  });
});
