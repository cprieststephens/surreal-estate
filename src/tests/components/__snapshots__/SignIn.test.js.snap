// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sign-in"
  >
    <form
      class="sign-in__form"
    >
      <div
        class="sign-in__item"
      >
        <label
          for="sign-in__email"
        >
          Email:
          <input
            class="sign-in__input"
            id="sign-in__email"
            placeholder="Enter email"
            required=""
            type="email"
          />
        </label>
      </div>
      <div
        class="sign-in__item"
      >
        <label
          for="sign-in__password"
        >
          Password:
          <input
            class="sign-in__input"
            id="sign-in__password"
            placeholder="Enter password"
            required=""
            type="password"
          />
        </label>
      </div>
      <button
        class="sign-in__button"
        type="button"
      >
        Sign In
      </button>
      <p
        class="sign-in__text"
      >
        Need to create an account?
        <span>
           
        </span>
        <a
          class="sign-in__link"
          href="/sign-up"
        >
          Sign Up
        </a>
      </p>
    </form>
    <div
      class="sign-in__alert"
    />
  </div>
</DocumentFragment>
`;
